### PR TITLE
auth/rpc: Take remote disk offline after maximum allowed attempts.

### LINKED ISSUE #3286: [distributed] rpc/client: Track errors upto 5 attempts and do not attempt Login() after.
## Expected Behavior
Currently if a disk is down rpcClient would attempt a dial for each such an attempt. Instead the remote disk should be remembered as dead unless explicitly toggled by healing or management API call etc.

## Current Behavior
Each login attempts and fails but net.Dial can be an expensive operation.

## Possible Solution
Remember the errors and then permanently take the disk offline after upto 5 attempts.

## Steps to Reproduce (for bugs)
Just take one of the server down.

## Context
Posix has a similar functionality of syscall.EIO errors, similar case should be added for rpcClient.

## Your Environment

* Version used: master branch.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Disks when are offline for a long period of time, we should
ignore the disk after trying Login upto 5 times.

This is to reduce the network chattiness, this also reduces
the overall time spent on `net.Dial`.

## Motivation and Context
Fixes #3286

## How Has This Been Tested?
Manually and through chaos test.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by harshavardhana** [comment]
Ping @krisis @krishnasrinivas can you review this?

File: cmd/auth-rpc-client.go [hunk 1]
```diff
@@ -127,30 +128,50 @@ func (authClient *AuthRPCClient) Close() error {
...
+	// if there is an error.
+	defer func() {
+		if err != nil {
+			authClient.serverIOErrCnt++
+		}
+	}()
+
+	// Take remote disk offline if the total server errors
+	// are more than maximum allowable IO error limit.
+	if authClient.serverIOErrCnt > maxAllowedIOError {
```
Comment (donatello) [inline cmd/auth-rpc-client.go]: Better to put this `if` block before calling the defer. If the threshold errors have been reached, then calling the defer it unnecessary.

**Review by donatello — APPROVED** [review]
LGTM - except for one minor suggested change.

File: cmd/auth-rpc-client.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/auth-rpc-client.go]: This is something that needs to be done for posix.go as well.  - Can you open a bug for this optimization?

**Pull request merged**